### PR TITLE
test: Refactor lint script to avoid globally mutable state

### PR DESCRIPTION
**Pull request by ExE-Boss**
Opened because of #3939.

review?(@Elchi3): This doesn’t currently require rebasing.

### CODE REVIEW

**Comment by Elchi3** [comment]
Thanks, this refactoring works for me. Did you also mean to include the proposal from 3939 here?

File: test/test-browsers.js [hunk 1]
```diff
@@ -66,7 +66,15 @@ function processData(data, displayBrowsers, requiredBrowsers, category, logger,
...
-    hasErrors |= processData(data[key], displayBrowsers, requiredBrowsers, category, logger, (path && path.length > 0) ? `${path}.${key}` : key);
+    hasErrors =
+      processData(
+        data[key],
+        displayBrowsers,
+        requiredBrowsers,
+        category,
+        logger,
+        path && path.length > 0 ? `${path}.${key}` : key,
+      ) || hasErrors;
```
Comment (ExE-Boss) [inline test/test-browsers.js]: This prevents `hasErrors` from being converted to a `number`.

File: test/test-browsers.js [hunk 2]
```diff
@@ -66,7 +66,16 @@ function processData(data, displayBrowsers, requiredBrowsers, category, logger,
...
+    hasErrors = processData(
+      data[key],
+      displayBrowsers,
+      requiredBrowsers,
+      category,
+      logger,
+      (path && path.length > 0)
+        ? `${path}.${key}`
+        : key,
+    ) || hasErrors;
```
Comment (ExE-Boss) [inline test/test-browsers.js]: This prevents `hasErrors` from being converted into&nbsp;a&nbsp;`number`.

Comment (Elchi3) [reply test/test-browsers.js]: Thanks, can we add that as a code comment? I think it would help understanding this line in the future.

**Pull request merged**